code/datums/flock/flockprotection.dm

```diff
@@ -1,76 +1,92 @@
-			var/obj/storage/closet/flock/flockcloset = source
-			if (!flockcloset.open)
-				boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
-				return TRUE
-		else if (istype(source, /obj/machinery/door/feather))
-			var/mob/living/critter/flock/drone/flockdrone = user
-			if (flockdrone.find_type_in_hand(/obj/item))
-				boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
-				return TRUE
-		else
+	var/we_take_those = FALSE
+	if (istype(source, /obj/lattice/flock) && isweldingtool(W) && W:try_weld(user,0,-1,0,0))
+		we_take_those = TRUE
+	else if (istype(source, /obj/storage/closet/flock))
+		var/obj/storage/closet/flock/flockcloset = source
+		if (!flockcloset.open)
+			if(!isflock(user) && istype(W, /obj/item/cargotele))
+				return
+			we_take_those = TRUE
+	else if (istype(source, /obj/machinery/door/feather))
+		if (user.equipped()) // Doors immedately pass attack_hand to Attackby, so this check is required.
+			we_take_those = TRUE
+	else
+		we_take_those = TRUE
+
+	if(we_take_those)
+		if(isflock(user) && src.flock_immune)
 			boutput(user, "<span class='alert'>The grip tool refuses to allow damage to this, jamming briefly.</span>")
 			return TRUE
-
-	if (!istype(user, /mob/living/critter/flock/drone) && report_obj_attack)
-		if (istype(source, /obj/lattice/flock))
-			if (isweldingtool(W) && W:try_weld(user,0,-1,0,0))
-				src.attempt_report_attack(source, user)
-		else if (istype(source, /obj/storage/closet/flock))
-			var/obj/storage/closet/flock/flockcloset = source
-			if (!flockcloset.open)
-				if (istype(W, /obj/item/cargotele) || istype(W, /obj/item/satchel))
-					return
-				else
-					src.attempt_report_attack(source, user)
-		else if (istype(source, /obj/machinery/door/feather))
-			if (user.equipped())
-				src.attempt_report_attack(source, user)
-		else
+		if(!isflock(user) && src.report_attack)
 			src.attempt_report_attack(source, user)
 
+/// Protect against someone chucking stuff at the parent.
+/datum/component/flock_protection/proc/handle_hitby_thrown(source, atom/hit_atom, datum/thrown_thing/thr)
+	var/mob/attacker = thr.user
+	if(!istype(attacker))
+		return
+	if (!isflock(attacker) && src.report_thrown)
+		src.attempt_report_attack(source, attacker)
+
+/// Protect against someone shooting the parent.
+/datum/component/flock_protection/proc/handle_hitby(source, obj/projectile/P)
+	var/mob/attacker = P.shooter
+	if(!istype(attacker))
+		return
+	if (!isflock(attacker) && src.report_proj)
+		src.attempt_report_attack(source, attacker)
+
+/// Look for flockdrones nearby who will snitch
 /datum/component/flock_protection/proc/attempt_report_attack(source, mob/attacker)
-	var/list/nearby_flockdrones = list()
+	var/mob/living/critter/flock/drone/snitch
 	for (var/mob/living/critter/flock/drone/flockdrone in view(7, source))
 		if (!isdead(flockdrone) && flockdrone.is_npc && flockdrone.flock)
```

Comment:
```suggestion
		if (!isdead(flockdrone) && flockdrone.is_npc && flockdrone.flock && flockdrone.flock == source.flock)
```

Reply:
Would not compile, source may not be flock. Probably should require a ref to the flock datum in the init

Reply:
Turns out all of the flock furniture does not keep a reference to the flock datum. Which means either I make it have one or we just assume flocks protect other flocks.

Reply:
Not sure if flock furniture needs it as it doesn't really have health? Think people deconstructing flock furniture could be viewed as a repurposing/replacement type of thing?

Reply:
Decoupled protection from flock being interested in protecting it. Enables multiple flocks and variant configuration of the hit detection